test/functional/controller/controller_suite_test.go

```diff
@@ -5,9 +5,18 @@ import (
 	. "github.com/onsi/gomega"
 
 	"testing"
+
+	"kubevirt.io/qe-tools/pkg/ginkgo-reporters"
 )
 
 func TestController(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "Controller Suite")
+	reporters := make([]Reporter, 0)
```

Comment:
can this block (and below) be in a global `BeforeEach`? I'm thinking that when we have a global framework object/struct then we can initialize it in a `BeforeEach` block (like done in k8s) and include multiple reporter registration at that point.

Reply:
I will take a look at it

Reply:
OK, I created Framework struct, but unfortunately no report was generated with following code

```golang
func NewFramework(baseName string) *Framework {
	f := &Framework{
		BaseName: baseName,
	}

	BeforeEach(f.BeforeEach)

	return f
}

func (f *Framework) BeforeEach() {
	f.Reporters = make([]Reporter, 0)
	if ginkgo_reporters.Polarion.Run {
		f.Reporters = append(f.Reporters, &ginkgo_reporters.Polarion)
	}
	if ginkgo_reporters.JunitOutput != "" {
		f.Reporters = append(f.Reporters, ginkgo_reporters.NewJunitReporter())
	}
}
```
So I ended up with current implementation. I am open to any suggestions.